src/Microsoft.ML.PCA/PcaTransformer.cs

```diff
@@ -596,6 +597,73 @@ private static void TransformFeatures(IExceptionContext ectx, in VBuffer<float>
 
                 dst = editor.Commit();
             }
+
+            public bool CanSaveOnnx(OnnxContext ctx) => true;
+
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                Host.CheckValue(ctx, nameof(ctx));
+
+                for (int i = 0; i < _numColumns; i++)
+                {
+                    var colPair = _parent.ColumnPairs[i];
+                    var transformInfo = _parent._transformInfos[i];
+                    string inputColumnName = colPair.inputColumnName;
+                    string outputColumnName = colPair.outputColumnName;
+                    if (!ctx.ContainsColumn(inputColumnName))
```

Comment:
>!ctx.ContainsColumn(inputColumnName) [](start = 24, length = 36)

Can this happen?
If we just continue when this happens, doesn't it mean that the ONNX model will be different than the original model? Shouldn't we throw in that case?

Reply:
I think that logic exists precisely to allow the higher level code to drop specific columns that it doesn't want in the model. The `RemoveVariablesInPipelineTest` test in `test\Microsoft.ML.Tests\OnnxConversionTest.cs` exercises that functionality where it drops specific columns in its call to `SaveOnnxCommand.ConvertTransformListToOnnxModel`.